src/axom/primal/tests/primal_rational_bezier.cpp

```diff
@@ -0,0 +1,560 @@
+
+    for(int i = 1; i < 128; i = i * 2)
+    {
+      curve.setWeight(1, i);
+
+      // linear for a coarse tolerance
+      {
+        const double tol = 0.1;
+        const double tol_sq = tol * tol;
+        EXPECT_TRUE(curve.isLinear(tol_sq));
+      }
+
+      // non-linear for a finer tolerance
+      {
+        const double tol = 0.01;
+        const double tol_sq = tol * tol;
+        EXPECT_FALSE(curve.isLinear(tol_sq));
+      }
+    }
+  }
+}
+
+TEST(primal_beziercurve, reverseOrientation)
+{
+  SLIC_INFO("Testing reverseOrientation() on Rational Bezier curves");
+
+  {
+    const int DIM = 2;
+    using CoordType = double;
+    using PointType = primal::Point<CoordType, DIM>;
+    using BezierCurveType = primal::BezierCurve<CoordType, DIM>;
+
+    // test different orders
+    for(int order = 0; order <= 10; ++order)
+    {
+      // control points for curve monotonically increase
+      axom::Array<PointType> pts(order + 1);
+      axom::Array<double> weights(order + 1);
+      for(int i = 0; i <= order; ++i)
+      {
+        pts[i] = PointType(i);
+        weights[i] = i;
+      }
+      BezierCurveType curve(pts.data(), weights.data(), order);
+
+      for(int i = 1; i <= order; ++i)
+      {
+        EXPECT_GT(curve[i][0], curve[i - 1][0]);
+      }
+
+      // create a reversed curve and check that it monotonically decreases
+      BezierCurveType reversed = curve;
+      reversed.reverseOrientation();
+
+      for(int i = 1; i <= order; ++i)
+      {
+        EXPECT_LT(reversed[i][0], reversed[i - 1][0]);
+      }
+
+      // Check that the control points are actually reversed
+      for(int i = 0; i <= order; ++i)
+      {
+        EXPECT_EQ(curve[i], reversed[order - i]);
+      }
+
+      // check that reversing again reverts to the original
+      BezierCurveType reversedAgain = reversed;
+      reversedAgain.reverseOrientation();
+      EXPECT_EQ(curve, reversedAgain);
+    }
+  }
+}
+
+TEST(primal_rationalbezier, winding_number)
+{
+  using Point2D = primal::Point<double, 2>;
+  using Bezier = primal::BezierCurve<double, 2>;
+  using CPolygon = primal::CurvedPolygon<double, 2>;
+
+  double abs_tol = 1e-8;
+  double lin_tol = 0;
+  double edge_tol = 0;
+
+  // Simple quarter circle shape
+  Point2D circle_nodes[] = {Point2D {1.0, 0.0},
+                            Point2D {1.0, 1.0},
+                            Point2D {0.0, 1.0}};
+  double weights[] = {2.0, 1.0, 1.0};
+  Bezier circle_arc(circle_nodes, weights, 2);
+
+  Point2D leg1_nodes[] = {Point2D {0.0, 1.0}, {0.0, 0.0}};
+  Bezier leg1(leg1_nodes, 1);
+
+  Point2D leg2_nodes[] = {Point2D {0.0, 0.0}, {1.0, 0.0}};
+  Bezier leg2(leg2_nodes, 1);
+
+  CPolygon quarter_circle;
+  quarter_circle.addEdge(circle_arc);
+  quarter_circle.addEdge(leg1);
+  quarter_circle.addEdge(leg2);
+
+  for(double theta = 0.01; theta < 1.5; theta += 0.05)
+    for(int i = 1; i < 9; i++)
+    {
+      double ri = 1.0 - std::pow(10, -i);
+      double ro = 1.0 + std::pow(10, -i);
```

Comment:
Minor suggestion (for readability):
```suggestion
      const double offset = std::pow(10, -i);
      const double ri = 1.0 - offset;
      const double ro = 1.0 + offset;
```